doc/vxlan/EVPN/EVPN_VxLAN_Multihoming.md

```diff
@@ -0,0 +1,1856 @@
+
+![](images/evpn_mh_diagram.PNG)
+
+__Figure 1: Typical EVPN Multihoming network__
+
+The devices H1, H4, and H5 are SH devices and connected to Vtep-1, Vtep-4, and Vtep-5 respectively. The devices H2 & H3 are multihomed and connected to EVPN VxLAN network with active-active redundancy using LAG. Device H2 is multihomed to Vtep-1 and Vtep-4 whereas device H3 is multihomed to Vtep-1, Vtep-2, Vtep-3, and Vtep-4.
+
+In order to configure EVPN Multihoming for the LAGs connecting multihomed devices, Type-0, Type-1, or Type-3 ESI is required to be configured on the LAG interfaces:
+- *T*ype-0: Operator configured ESI*
+- *Type-1: ESI automatically derived from LACP partner MAC*
+- *Type-3: ESI automatically derived from configured LAG system-mac address*
+
+For a given Multihomed Ethernet-Segment, the value of ES-ID should be same on all of the participating VTEPs.
+
+For simplicity, assume Type-3 ESI is configured on the PortChannel interfaces shown in the above diagram. The Vtep-1 and Vtep-4 LAG configuration will look like this:
+
+```
+sudo config interface sys-mac add PortChannel1 00:00:00:00:11:11
+sudo config interface evpn-esi add PortChannel1 auto-system-mac
+
+sudo config interface sys-mac add PortChannel2 00:00:00:00:22:22
+sudo config interface evpn-esi add PortChannel2 auto-system-mac
+```
+
+Whereas, on Vtep-2 and Vtep-3 only PortChannel2 config is required since only PortChannel2 is multihomed to these VTEPs.
+```
+sudo config interface sys-mac add PortChannel2 00:00:00:00:22:22
+sudo config interface evpn-esi add PortChannel2 auto-system-mac
+```
+The Type-3 ESI value will be derived by combining the configured system-mac address on the PortChannel interface, and the PortChannel number. The system-mac is required to be same on all of the VTEPs multihoming a given LAG.
+
+The above diagram and PortChannel configurations will be used in subsequent sub-sections to explain the traffic forwarding scenarios in EVPN Multihoming network.
+
+#### 1.2.1.1 BUM forwarding in EVPN MH
+The BUM (Broadcast, Unknown Unicast, and Multicast) traffic in EVPN Multihoming network requires different handling depending on whether BUM traffic is originated from VTEP participating or not participating in the EVPN Multihoming.
+
+##### 1.2.1.1.1 Local-bias and split-horizon filtering
+In multi-homed scenario, if the BUM traffic originates from a device attached to an EVPN Ethernet-Segment, local-bias procedure is applied and BUM traffic is flooded on all locally attached multi-homed and single-homed devices. The BUM traffic is also replicated to the remote VTEPs, and split-horizon filtering is applied such that traffic is not replicated to any of the devices that are attached to the shared EVPN Ethernet-Segment.
+
+Local-bias procedure is described in EVPN VxLAN Overlay [RFC 8365](https://datatracker.ietf.org/doc/html/rfc8365). The behavior of local-bias is explained with the help of below diagram.
```

Comment:
@helloanandhi , No Vxlan-gpe right now. This hld can be extended in future for gpe. There is no bum bit in original vxlan header.

Reply:
@hasan-brcm  If VXLAN-GPE is not supported, does this mean that ingress replication cannot be used in the current implementation due to BUM traffic duplication that described in the [https://datatracker.ietf.org/doc/html/rfc8365#section-8.3.3](https://datatracker.ietf.org/doc/html/rfc8365#section-8.3.3)

Reply:
@gord1306 , VXLAN-GPE is not a mandate as per RFC8365. It is just a suggestion to avoid (low probability) transient loop.
> If it is desired to
 avoid occurrence of such transient packet duplication (however low
 probability that may be), then VXLAN-GPE encapsulation needs to be
 used between these PEs and the ingress PE needs to set the BUM
 Traffic Bit (B bit) to indicate that this is an ingress-
replicated BUM traffic.